core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -467,6 +467,15 @@ private[deploy] class Worker(
         }.foreach { dir =>
           logInfo(s"Removing directory: ${dir.getPath}")
           Utils.deleteRecursively(dir)
+
+          // Remove some registeredExecutors information of DB in external shuffle service when
+          // #spark.shuffle.service.db.enabled=true, the one which comes to mind is, what happens
+          // if an application is stopped while the external shuffle service is down?
+          // So then it'll leave an entry in the DB and the entry should be removed.
+          if (conf.get(config.SHUFFLE_SERVICE_DB_ENABLED) &&
+            conf.get(config.SHUFFLE_SERVICE_ENABLED)) {
```

Comment:
nit: double indent this line (4 spaces)

Reply:
sorry , i don't understand why this is necessary.

Reply:
its a continuation of the condition of the `if`.   That helps separate it from the body, which is only indented 2 spaces.  eg. like this:

https://github.com/apache/spark/blob/190a3a4ad8e648d4ed4b38c5189b3baf75b1fc52/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/AnalysisHelper.scala#L133-L137